### ISSUE TITLE
Miscalculated evals from wrapping text

### ISSUE THREAD

**Issue by yaxu** [seansay/atom-tidal#9]
If you shrink the window so that a line of code wraps, then try to evaluate a pattern after it, it miscounts the line you're on and evaluates the line after it.

For example if you were to try to evaluate `d2` here then `d3` would be evaluated:

````haskell
d1 $ very long pattern goes here alwiejf awleijga wleigja welgi jwaegliwa jeglwa ijegalwei jglwaeigjawel gijawelgi jawelgi j

d2 $ sound "bd sn"
d3 $ sound "glitch glitch:2"
````

**Comment by seansay** [comment]
current I'm using `getCurrentParagraphBufferRange`, which is not very smart of getting the part of haskell code I think. need to use some method related to grammar scope

**Comment by seansay** [comment]
not sure how to do select  region like emacs's `region-beginning` and `region-end`

**Comment by bgold-cosmos** [comment]
I've noticed this too, it really seems to be a problem with Atom's `getCurrentParagraphBufferRange`.  There's also a problem where a "paragraph" won't include an ending `]` sitting on a line by itself.  I tried to find a workaround but haven't come up with anything yet.

**Comment by seansay** [comment]
@bgold-cosmos 
yes, @yaxu is posting an issue in atom core , hope they will fix this soon! #11

**Comment by seansay** [comment]
@yaxu 
I post a pr here, they're merged. The most recent release fix this problem
